test/micro/org/openjdk/bench/jdk/incubator/vector/MatrixMultiplicationBenchmark.java

```diff
@@ -0,0 +1,184 @@
+//
+// Copyright (c) 2003, 2023, Oracle and/or its affiliates. All rights reserved.
+// DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+//
+// This code is free software; you can redistribute it and/or modify it
+// under the terms of the GNU General Public License version 2 only, as
+// published by the Free Software Foundation.
+//
+// This code is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// version 2 for more details (a copy is included in the LICENSE file that
+// accompanied this code).
+//
+// You should have received a copy of the GNU General Public License version
+// 2 along with this work; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+// or visit www.oracle.com if you need additional information or have any
+// questions.
+//
+//
+package org.openjdk.bench.jdk.incubator.vector;
+
+import java.util.Random;
+import jdk.incubator.vector.*;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.Blackhole;
+
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+@Fork(value = 1, jvmArgsPrepend = {"--add-modules=jdk.incubator.vector", "-XX:-TieredCompilation"})
+public class MatrixMultiplicationBenchmark{
+
+    private static final VectorSpecies<Float> SPECIES = FloatVector.SPECIES_PREFERRED;
+
+    private final int BLOCK_SIZE = 16;
+    @Param({"32", "1024"})
+    private int size;
+    private float[] left;
+    private float[] right;
+
+    private float[] result;
+
+    @Setup
+    public void setup(){
+        this.left = MatrixMultiplicationBenchmark.newFloatRowMatrix(size * size);
+        this.right = MatrixMultiplicationBenchmark.newFloatRowMatrix(size * size);
+        this.result = new float[size * size];
+
+    }
+
+    @Benchmark
+    public void mmulBaseline(){
+        baseline(left, right, result, size);
```

Comment:
```suggestion
    public float[] mmulBaseline(){
        return baseline(left, right, result, size);
```
Same for other methods